src/waveform/renderers/waveformmarkrange.cpp

```diff
@@ -43,6 +27,11 @@ WaveformMarkRange::WaveformMarkRange(
         DEBUG_ASSERT(!m_markEndPointControl); // has not been created yet
         m_markEndPointControl = std::make_unique<ControlProxy>(group, endControl);
     }
+    QString rangeName = startControl;
+    if (startControl.contains(QStringLiteral("_start_position"))) {
```

Comment:
this parses the string twice.

Reply:
This is only used in the error case, so I would duplicate it and move it closer to usage.
This way we are sure it is only executed when really needed.

Reply:
:+1:  Done.